### PR TITLE
feat(obs): import obs resource and add unit test and docs for 10sp2 p…

### PR DESCRIPTION
**Pull request by Zhukun-Huawei**
…roject

**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./flexibleengine' TESTARGS='-run=TestAccSomethingV0_basic'
...
=== RUN   TestAccSomethingV0_basic
--- PASS: TestAccSomethingV0_basic (70.75s)
PASS
ok      github.com/FlexibleEngineCloud/terraform-provider-flexibleengine/flexibleengine       70.796s
```

### CODE REVIEW

File: docs/data-sources/obs_buckets.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The ID of the list.
+
+* `buckets` - A list of OBS buckets.
+
+The `buckets` block supports:
```
Comment (Zippo-Wang) [inline docs/data-sources/obs_buckets.md:34]:
```markdown
* `buckets` - A list of OBS buckets. The [buckets](#obs_buckets) object structure is documented below.

<a name="obs_buckets"></a>
The `buckets` block supports:
```

This specification is suitable to the fields which have son-fields.

File: docs/resources/obs_bucket_object_acl.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+* `bucket` - (Required, String, ForceNew) Specifies the name of the bucket which the object belongs to.
+
+  Changing this parameter will create a new resource.
+
+* `key` - (Required, String, ForceNew) Specifies the name of the object to which to set the acl.
+
+  Changing this parameter will create a new resource.
+
+* `public_permission` - (Optional, List) Specifies the object public permission.
+  The [permission_struct](#OBSBucketObjectAcl_permission_struct) structure is documented below.
```
Comment (Zippo-Wang) [inline docs/resources/obs_bucket_object_acl.md:61]:
```
The [public_permission](#OBSBucketObjectAcl_permission_struct) structure is documented below.
```

Advice, The name in middle bracket should be same to field name

File: docs/resources/obs_bucket_object_acl.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+* `account_permission` - (Optional, List) Specifies the object account permissions.
+  The [account_permission_struct](#OBSBucketObjectAcl_account_permission_struct) structure is documented below.
```
Comment (Zippo-Wang) [inline docs/resources/obs_bucket_object_acl.md:64]:
```markdown
The [account_permission](#OBSBucketObjectAcl_account_permission_struct) structure is documented below.
```

Advice, The name in middle bracket should be same to field name

File: docs/resources/obs_bucket_object_acl.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+<a name="OBSBucketObjectAcl_permission_struct"></a>
+The `permission_struct` block supports:
```
Comment (Zippo-Wang) [inline docs/resources/obs_bucket_object_acl.md:67]:
```markdown
The `public_permission` block supports:
```

Advice, this name should be same to field name

File: docs/resources/obs_bucket_object_acl.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+* `access_to_object` - (Optional, List) Specifies the access to object. Only **READ** supported.
+
+* `access_to_acl` - (Optional, List) Specifies the access to acl. Valid values are **READ_ACP** and **WRITE_ACP**.
+
+<a name="OBSBucketObjectAcl_account_permission_struct"></a>
+The `account_permission_struct` block supports:
```
Comment (Zippo-Wang) [inline docs/resources/obs_bucket_object_acl.md:74]:
```md
The `account_permission` block supports:
```
Advice, this name should be same to field name

File: docs/resources/obs_bucket_object_acl.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+* `access_to_acl` - (Optional, List) Specifies the access to acl. Valid values are **READ_ACP** and **WRITE_ACP**.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The name of the bucket object key.
+* `owner_permission` - The object owner permission information.
+  The [owner_permission_struct](#OBSBucketObjectAcl_owner_permission_struct) structure is documented below.
```
Comment (Zippo-Wang) [inline docs/resources/obs_bucket_object_acl.md:89]:
```md
The [owner_permission](#OBSBucketObjectAcl_owner_permission_struct) structure is documented below.
```
Advice, The name in middle bracket should be same to field name

File: docs/resources/obs_bucket_object_acl.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+<a name="OBSBucketObjectAcl_owner_permission_struct"></a>
+The `owner_permission_struct` block supports:
```
Comment (Zippo-Wang) [inline docs/resources/obs_bucket_object_acl.md:92]:
```md
The `owner_permission` block supports:
```
Advice, this name should be same to field name

File: docs/resources/obs_bucket_policy.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the OBS bucket policy resource. If omitted, the
+  provider-level region will be used. Changing this creates a new OBS bucket policy resource.
+
+* `bucket` - (Required, String, ForceNew) Specifies the name of the bucket to which to apply the policy.
```
Comment (Zippo-Wang) [inline docs/resources/obs_bucket_policy.md:76]:
```md
* `bucket` - (Required, String, ForceNew) Specifies the name of the bucket to which to apply the policy.

  Changing this creates a new OBS bucket policy resource.
```
lost `Changin this ... resource`

File: flexibleengine/acceptance/resource_flexibleengine_obs_bucket_object_acl_test.go [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+	return fmt.Sprintf(`
+%s
+
+resource "flexibleengine_obs_bucket_object_acl" "test" {
+  bucket = flexibleengine_obs_bucket.bucket.bucket
+  key    = flexibleengine_obs_bucket_object.object.key
+
+  account_permission {
+    access_to_acl    = ["READ_ACP"]
+    account_id       = "1000010022"
```
Comment (Zippo-Wang) [inline flexibleengine/acceptance/resource_flexibleengine_obs_bucket_object_acl_test.go:169]:
```go
    access_to_acl = ["READ_ACP"]
    account_id    = "1000010022"
```

two many spaces

File: flexibleengine/acceptance/resource_flexibleengine_obs_bucket_object_acl_test.go [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+	return fmt.Sprintf(`
+%s
+
+resource "flexibleengine_obs_bucket_object_acl" "test" {
+  bucket = flexibleengine_obs_bucket.bucket.bucket
+  key    = flexibleengine_obs_bucket_object.object.key
+
+  account_permission {
+    access_to_acl    = ["READ_ACP"]
+    account_id       = "%s"
```
Comment (Zippo-Wang) [inline flexibleengine/acceptance/resource_flexibleengine_obs_bucket_object_acl_test.go:189]:
```go
    access_to_acl = ["READ_ACP"]
    account_id    = "%s"
```

two many spaces

File: flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+
+		return fmt.Sprintf("%s/s3", policyRes.Primary.ID), nil
+	}
+}
+
+func testAccObsBucketPolicyConfig(bucketName string) string {
+	return fmt.Sprintf(`
+resource "flexibleengine_obs_bucket" "bucket" {
+	bucket = "%s"
+}
```
Comment (Zippo-Wang) [inline flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go:226]:
```
resource "flexibleengine_obs_bucket" "bucket" {
  bucket = "%s"
}
```

use two space instead TAB

File: flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+
+resource "flexibleengine_obs_bucket_policy" "policy" {
+	bucket = flexibleengine_obs_bucket.bucket.bucket
+	policy =<<POLICY
+{
```
Comment (Zippo-Wang) [inline flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go:231]: ditto

File: flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go [hunk 13]
```diff
@@ -0,0 +1,301 @@
...
+}
+POLICY
+}
+`, bucketName, bucketName)
+}
+
+func testAccObsBucketPolicyConfig_updated(bucketName string) string {
+	return fmt.Sprintf(`
+resource "flexibleengine_obs_bucket" "bucket" {
+	bucket = "%s"
```
Comment (Zippo-Wang) [inline flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go:250]: ditto

File: flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go [hunk 14]
```diff
@@ -0,0 +1,301 @@
...
+}
+
+resource "flexibleengine_obs_bucket_policy" "policy" {
+	bucket = flexibleengine_obs_bucket.bucket.bucket
```
Comment (Zippo-Wang) [inline flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go:254]: ditto

File: flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go [hunk 15]
```diff
@@ -0,0 +1,301 @@
...
+}
+POLICY
+}
+`, bucketName, bucketName)
+}
+
+func testAccObsBucketPolicyS3Foramt(bucketName string) string {
+	return fmt.Sprintf(`
+resource "flexibleengine_obs_bucket" "bucket" {
+	bucket = "%s"
```
Comment (Zippo-Wang) [inline flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go:275]: ditto

File: flexibleengine/provider.go [hunk 16]
```diff
@@ -320,6 +320,8 @@ func Provider() *schema.Provider {
 			"flexibleengine_modelarts_datasets":         modelarts.DataSourceDatasets(),
 			"flexibleengine_modelarts_dataset_versions": modelarts.DataSourceDatasetVerions(),
 
+			"flexibleengine_obs_bucket_object": obs.DataSourceObsBucketObject(),
+
```
Comment (Zippo-Wang) [inline flexibleengine/provider.go:324]: why provider.go have only one data source name? this field should have two resource and two data source

File: docs/data-sources/obs_buckets.md (hunk 1 above)
Comment (Zhukun-Huawei) [reply docs/data-sources/obs_buckets.md]: image
revised

File: docs/resources/obs_bucket_object_acl.md (hunk 2 above)
Comment (Zhukun-Huawei) [reply docs/resources/obs_bucket_object_acl.md]: image
revised

File: docs/resources/obs_bucket_object_acl.md (hunk 4 above)
Comment (Zhukun-Huawei) [reply docs/resources/obs_bucket_object_acl.md]: image
revised

File: docs/resources/obs_bucket_policy.md (hunk 8 above)
Comment (Zhukun-Huawei) [reply docs/resources/obs_bucket_policy.md:76]: image
revised

File: flexibleengine/acceptance/resource_flexibleengine_obs_bucket_object_acl_test.go (hunk 9 above)
Comment (Zhukun-Huawei) [reply flexibleengine/acceptance/resource_flexibleengine_obs_bucket_object_acl_test.go]: image
revised

File: flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go (hunk 11 above)
Comment (Zhukun-Huawei) [reply flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go]: image
revised

File: flexibleengine/provider.go (hunk 16 above)
Comment (Zhukun-Huawei) [reply flexibleengine/provider.go:332]: image
good catch, I lose these resources or data sources accidently,  and I check the ut results again, the results are right.

File: flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go (hunk 12 above)
Comment (Zhukun-Huawei) [reply flexibleengine/acceptance/resource_flexibleengine_obs_bucket_policy_test.go]: image
revised

**Pull request merged**